numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -3398,20 +3405,15 @@ def host_to_device(dst, src, size, stream=0):
     it should not be changed until the operation which can be asynchronous
     completes.
     """
-    varargs = []
+    fn = driver.cuMemcpyHtoD
+    args = (device_pointer(dst), host_pointer(src, readonly=True), size)
 
     if stream:
-        assert isinstance(stream, Stream)
+        assert isinstance(stream, (Stream, ExperimentalStream))
         fn = driver.cuMemcpyHtoDAsync
-        if USE_NV_BINDING:
-            handle = stream.handle.value
-        else:
-            handle = stream.handle
-        varargs.append(handle)
-    else:
-        fn = driver.cuMemcpyHtoD
+        args += (_stream_handle(stream),)
 
-    fn(device_pointer(dst), host_pointer(src, readonly=True), size, *varargs)
+    fn(*args)
 
 
 def device_to_host(dst, src, size, stream=0):
```

Comment:
As mentioned below (or above), `stream` semantics is changed which probably has a bigger impact to this method, because the copy is now asynchronous and to access `src` on host a stream synchronization is needed.